module/icp/core/kcf_mech_tabs.c

```diff
@@ -664,6 +664,9 @@ kcf_remove_mech_provider(char *mech_name, kcf_provider_desc_t *prov_desc)
 
 	mutex_exit(&mech_entry->me_mutex);
 
+	if (prov_mech == NULL)
+		return;
+
```

Comment:
What about moving this in to the `default:` case instead of a break.  Something like:

```
        /* unexpected crypto_provider_type_t */
        mutex_exit(&mech_entry->me_mutex);
        return;
```